packages/blade/src/components/PreviewWindow/_decision/decisions.md

```diff
@@ -0,0 +1,144 @@
+| `onDragChange`   | `(position: { x: number; y: number }) => void`         | —       | No       | Callback invoked when the drag position changes.                            |
+| `zoomScaleStep`       | `number`                                               | `10`    | No       | A number between 5 and 30 that determines the zoom step per click.         |
+| `defaultZoom`    | `number`                                               | `10`    | No       | A number between 1 and 100 that sets the initial zoom level.               |
+
+```tsx
+type PreviewWindowProps = {
+  children: React.ReactElement;
+  onFullScreen?: () => void;
+  isDragEnabled?: boolean;
+  zoom: number;
+  onZoomChange?: (newZoom: number) => void;
+  onDragChange?: (position: { x: number; y: number }) => void;
+  defaultZoom?: number;
+  zoomScaleStep?: number;
+};
+```
+
+
+In addition, we will have three layout components:
+ - PreviewHeader
+ - PreviewBody
+ - PreviewFooter
+
+
+```tsx
+type PreviewHeader = {
+  title?: string;
+  trailing?: React.ReactElement;
+};
+
+type PreviewBody = {
+  children: React.ReactElement;
+};
+
+type PreviewFooter = {
+  trailing?: React.ReactElement;
+  showZoomPercentage?: boolean;
+};
+```
+
+Example Usage - 
+
+```tsx
+<PreviewWindow>
+  <PreviewHeader title="Preview" trailing={<DownloadButton />} />
+  <PreviewBody>
+  <img src="/preview.jpg" alt="Example Preview" />
+  </PreviewBody>
+  <PreviewFooter trailing={<DeviceToggle />} />
+</PreviewWindow>
+
+// Without zoom controls
+<PreviewWindow
+  additionalPreviewControls={<CustomComponent />}
+>
+  <PreviewBody>
+  <ComponentToPreview />
+  </PreviewBody>
+</PreviewWindow>
+
+// Without Drag
+<PreviewWindow isDragEnabled={false}>
+  <StaticComponent />
+</PreviewWindow>
+```
+
+## Alternative API
+- For PreviewWindow component
+    ```tsx
+    // Using prop-driven content instead of children
+    <PreviewWindow content={<ImageComponent />} headingText="Preview" />
+    ```
+- For `additionalControls `
+
+    we can provide a wrapper components  like `<Header/>` and  `<Footer/>` and consumers can pass their custom controls , we will add zoom reset automatically.
+    ```tsx
+    <PreviewPanel>
+    <Header>
+     <Controls/>
+    </Header>
+    <Footer>
+         <Controls/>
+    </Footer>
+    ```
+
+    we can also have a hook `usePreviewWindow` which will expose methods like `onZoomIncrementButtonClick` ,  `onZoomDecrementButtonClick` , `onFullScreenClick` , `onResetZoomResetClick` and consumers can use it like 
+    ``` tsx
+    <PreviewPanel>
+      <Header>
+        <FullScreenButton onClick={onFullScreenClick} />
+        <CustomComponent/>
+      </Header>
+      <PreviewFile/>
+      <Footer>
+       <ZoomIncrementButton onClick={onZoomIncrementButtonClick} />
+       <ZoomDecrementButton onClick={onZoomDecrementButtonClick} />
+      </Footer>
+    </PreviewPanel>
+    ```
+
+
```

Comment:
Add relevant API considerations / summary MoMs of the discussions that we had. Like integration with third party library, how will consumers use this etc etc.